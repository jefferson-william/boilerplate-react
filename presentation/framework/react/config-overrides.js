const rewireReactHotLoader = require('react-app-rewire-hot-loader')
const { override, removeModuleScopePlugin } = require('customize-cra')
const { paths } = require('react-app-rewired')
const rewireAliases = require('react-app-rewire-aliases')
const WebpackAssetsManifest = require('webpack-assets-manifest')
const path = require('path')

// const multipleEntry = require('react-app-rewire-multiple-entry')([
//   {
//     entry: `${paths.appSrc}/components/Button/index.tsx`,
//   },
// ])

module.exports = override((config, env) => {
  let configuration = { ...config }

  removeModuleScopePlugin()(configuration)

  configuration = rewireAliases.aliasesOptions({
    '@': path.resolve(__dirname, `${paths.appPublic}`),
    '~': path.resolve(__dirname, `${paths.appSrc}`),
  })(config, env)

  if (env === 'development') {
    configuration.resolve.alias['react-dom'] = '@hot-loader/react-dom'
  }

  configuration = rewireReactHotLoader(configuration, env)

  configuration.plugins.push(
    new WebpackAssetsManifest({
      output: 'assets-manifest.json',
      // customize(entry, original, manifest, asset) {
      //   return {
      //     key: entry.key,
      //     value: entry.value,
      //   }
      // },
      // transform(assets, manifest) {
      //   return {
      //     files: assets,
      //   }
      // },
    }),
  )

  // multipleEntry.addMultiEntry(configuration)
  // console.log(configuration)
  // throw new Error()

  return configuration
})
