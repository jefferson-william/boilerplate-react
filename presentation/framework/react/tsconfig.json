{
  "extends": "./tspaths.json",
  "compilerOptions": {},
  "include": [
    "src"
  ]
}
