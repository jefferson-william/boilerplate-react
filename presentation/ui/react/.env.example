PORT=4000
PUBLIC_URL=http://localhost:4000
REACT_APP_VENDOR_URL=http://localhost:3000
SKIP_PREFLIGHT_CHECK=true
