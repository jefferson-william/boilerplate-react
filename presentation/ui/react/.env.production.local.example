PORT=6000
PUBLIC_URL=http://localhost:6000
REACT_APP_VENDOR_URL=http://localhost:5000
SKIP_PREFLIGHT_CHECK=true
