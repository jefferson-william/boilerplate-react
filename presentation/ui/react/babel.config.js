module.exports = {
  presets: ['@babel/preset-env', '@babel/preset-react', '@babel/preset-typescript'],
  plugins: [
    'react-hot-loader/babel',
    '@babel/plugin-transform-modules-systemjs',
    ['@babel/plugin-proposal-private-methods', { loose: true }],
    [
      'babel-plugin-styled-components',
      {
        ssr: false,
        displayName: true,
        filename: true,
      },
    ],
    [
      'babel-plugin-import',
      {
        libraryName: '@material-ui/core',
        libraryDirectory: '',
        camel2DashComponentName: false,
      },
      'core',
    ],
    [
      'module-resolver',
      {
        root: ['./'],
        alias: {
          '~': './src',
          '@': './public',
        },
      },
    ],
  ],
}
