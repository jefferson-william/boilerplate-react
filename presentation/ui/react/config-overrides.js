const rewireReactHotLoader = require('react-app-rewire-hot-loader')
const { paths } = require('react-app-rewired')
const rewireAliases = require('react-app-rewire-aliases')
const reactAppRewireBuildDev = require('react-app-rewire-build-dev')
const path = require('path')

module.exports = {
  webpack(config, env) {
    let configuration = rewireAliases.aliasesOptions({
      '@': path.resolve(__dirname, `${paths.appPublic}`),
      '~': path.resolve(__dirname, `${paths.appSrc}`),
    })(config, env)

    if (env === 'development') {
      configuration.resolve.alias['react-dom'] = '@hot-loader/react-dom'
    }

    configuration = rewireReactHotLoader(configuration, env)

    configuration = reactAppRewireBuildDev(config, env, {
      outputPath: 'build',
      basename: 'static',
      hotReloadPort: 9000,
    })

    configuration.output.futureEmitAssets = false
    configuration.output.jsonpFunction = '@react'
    configuration.output.libraryTarget = 'system'
    configuration.externals = {
      rxjs: 'rxjs',
    }

    return configuration
  },
  devServer(configFunction) {
    return (proxy, allowedHost) => {
      const config = configFunction(proxy, allowedHost)

      config.disableHostCheck = true
      config.sockHost = 'localhost'
      config.sockPort = process.env.PORT
      config.headers = {
        'Access-Control-Allow-Origin': '*',
      }

      return config
    }
  },
}
