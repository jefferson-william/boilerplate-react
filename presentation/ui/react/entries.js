module.exports = {
  button: './src/components/Button/single.spa.tsx',
  text: './src/components/Text/single.spa.tsx',
}
