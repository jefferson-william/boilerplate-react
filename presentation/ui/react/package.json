{
  "name": "@app/react",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "dev": "echo none",
    "start": "echo none",
    "build": "echo none",
    "lint": "echo none"
  },
  "dependencies": {
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-i18next": "11.11.4",
    "react-scripts": "4.0.3",
    "styled-components": "5.3.0",
    "single-spa-react": "^4.1.0",
    "systemjs-webpack-interop": "2.3.7",
    "typescript": "^4.1.2"
  },
  "devDependencies": {
    "@types/jest": "^26.0.15",
    "@types/node": "^12.0.0",
    "@types/react": "^17.0.0",
    "@types/react-dom": "^17.0.0",
    "@types/styled-components": "5.1.11",
    "craco-plugin-react-hot-reload": "0.1.0",
    "craco-plugin-single-spa-application": "^1.0.6",
    "msw": "0.33.2"
  }
}
