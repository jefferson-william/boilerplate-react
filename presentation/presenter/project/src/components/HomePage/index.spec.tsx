import { render, screen } from '@app/react/test'
import { HomePage } from './index'

test('renders learn react link', () => {
  const [wrapper] = render(<HomePage />)

  const linkElement = screen.getByText(/material-ui/i)

  expect(linkElement).toBeInTheDocument()

  expect(wrapper.asFragment()).toMatchSnapshot()
})
