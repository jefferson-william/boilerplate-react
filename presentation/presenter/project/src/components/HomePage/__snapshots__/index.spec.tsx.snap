// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <div
      class="home"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root home__menu-button MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <h6
            class="MuiTypography-root home__title MuiTypography-h6 MuiTypography-noWrap"
          >
            Material-UI
          </h6>
        </div>
      </header>
      <main
        class="home__main"
      >
        <p>
          Edit and save to reload.
        </p>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
